docs/programming-oak.md

```diff
@@ -379,20 +379,15 @@ each chat room and saves off the write handle that will be used to send messages
 to the room:
 
 <!-- prettier-ignore-start -->
-[embedmd]:# (../examples/chat/module_0/rust/src/lib.rs Rust /.*Create a new Node/ /^ +\);$/)
+[embedmd]:# (../examples/chat/module_0/rust/src/lib.rs Rust /.*channel_create\(\)/ /\}$/)
 ```Rust
-        // Create a new Node for this room.
         let (wh, rh) = oak::channel_create().unwrap();
         oak::node_create("room-config", rh);
         oak::channel_close(rh.handle);
-
-        self.rooms.insert(
-            req.admin_id.clone(),
-            Room {
-                name: req.name,
-                channel: wh,
-            },
-        );
+        Room {
+            channel: wh,
+            admin_token,
+        }
```

Comment:
I still find this code a bit cumbersome :(
- I do get why we use halfs, but maybe we need to encapsulate that better? I don't find it at all natural that the function called channel_create returns a tuple and that I have to reason about members separately. 
- If the pattern is to create node with rh and then close it, why not encapsulate that?

Reply:
- Yes, I think we're expecting higher layers of abstraction to appear in the SDK to hide these details.  But writing a few examples helps us figure out the patterns that are best to abstract.
 - The pattern of {create channel, create node with read half, drop local read half} is definitely a candidate for one of those higher level helpers.
 - "room-config" is the name of the config that describes a per-Room Wasm Node (the config currently holds the Wasm module bytes for the Node, and [may soon](https://github.com/project-oak/oak/pull/464) also hold the name of the main entrypoint)

Reply:
> "room-config" is the name of the config that describes a per-Room Wasm Node 

Should we have this strongly typed? I don't like passing strings :)

Reply:
> higher layers of abstraction to appear in the SDK 

I don't really understand the need for this PR in that case :)

Reply:
> > higher layers of abstraction to appear in the SDK
> 
> I don't really understand the need for this PR in that case :)

It's a prototype for how we want those higher layer to look like. We could have done it in a separate example, but I thought it would be more useful to operate on an existing example. It sounds like I was wrong :) I'll try and do experiments in separate, perhaps more focused examples in the future, if it helps, e.g. create an IPC example and only hack on IPC stuff there. Would this help?

Reply:
> Should we have this strongly typed? I don't like passing strings :)

IIRC this is passed into the oak runtime via protobuf'ed ApplicationConfiguration which specifies the names as strings. Of course the SDK can provide a strongly typed interface here, for the pseudo nodes it would be easy, for developer created nodes it would require boiler plate generation, but may be possible.